modules/core/src/lib/api/parse.ts

```diff
@@ -62,6 +63,11 @@ export async function parse(
 async function parseWithLoader(loader, data, options, context) {
   validateWorkerVersion(loader);
 
+  if (isResponse(data)) {
```

Comment:
We also have a `makeResponse` function.  IIRC it is already called in load ensuring that we have often have response object available here.

Reply:
But we can't rely on it - in any case the types allows data to be an arbitrary `DataType` to be passed to this function, so best to be safe. Appending the underlying Response where valid seems reasonable